api/envoy/config/trace/v2/trace.proto

```diff
@@ -80,6 +80,17 @@ message ZipkinConfig {
   // Determines whether client and server spans will shared the same span id.
   // The default value is true.
   google.protobuf.BoolValue shared_span_context = 4;
+
+  // Available zipkin collector endpoint versions.
+  enum CollectorEndpointVersion {
+    DEFAULT = 0;
```

Comment:
Not sure about this one. I know `json v2` is the popular one but I would prefer this to be a conscious decision from user side.

Reply:
Hum, yeah. To clarify, this is a way for us to deprecate v1 later since the previous one doesn't require to specify the version.

Reply:
True. We have the mission of not having string concatenation when building the json. I wish that will make the code better in terms of maintainability.

Probably, I can have everything here, i.e. `HTTP_JSON_V1`, `HTTP_JSON`, `HTTP_PROTO` and `GRPC` and make the `HTTP_JSON_V1` as default plus hiding the unimplemented ones.

Reply:
If json must happen via a protobuf libary as opposed to another library, one way out is to make a proto that matches the json format. the syntax is almost the same except hex instead of bytes in ids and literals instead of bytes in IPs. I'd hope the actual mission is getting this to support json, and that's one way to do it.

Reply:
For example, I think almost everyone ask directly for json v2. we don't solve that by sending proto.. we only solve the problem of giving an alterative to json v1, but one many cannot use because most data pipelines use json v2 (ex netflix, yelp etc). tools written against proto are usually not the same tools written against json, so if we move to proto, we need the tools ingesting to then convert proto to json, which is easier, but still.. I think you get what I mean right?

Reply:
https://github.com/openzipkin/zipkin-api/pull/77/files

Reply:
I would also prefer an `UNKNOWN` here or `NOT_SET` if there is no standard default.

Reply:
Set it to `NOT_SET`.